### ISSUE TITLE
How to pull training (or validation) loss during training?

### ISSUE THREAD

**Issue by lotsoffood** [awslabs/gluon-ts#549]
I responded to `https://github.com/awslabs/gluon-ts/issues/258` but though I should open a new question on this.

Is there any way to pull training loss and validation loss (once implemented) during training? I utilize Sacred for tracking experiments, and they have a function ex.log_scalar('your_metric', value, step) where step would be the epoch, and value the training or validation loss, and is logged each time an epoch finishes. Would love to be able to use Sacred/Omniboard to visualize.

**Comment by mbohlkeschneider** [comment]
Hi @lotsoffood, 

I don't really know how Sacred works so I cannot help with that. The losses are logged [here](https://github.com/awslabs/gluon-ts/blob/master/src/gluonts/trainer/_base.py#L277). 

What do you mean with "pulling the loss"?

**Comment by lotsoffood** [comment]
@mbohlkeschneider This is what I'm looking for!

Sacred has a method for their experiment object, `ex.log_scalar()` that allows you to log some kind of loss at each step of the way. I would just insert that into the location you linked to. "pulling the loss" == get access to the loss outside of the `Trainer` object.

**Comment by mbohlkeschneider** [comment]
Glad that helped. Closing this for now, please reopen if you have further questions.
